internal/github/repositories.go

```diff
@@ -0,0 +1,66 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/shurcooL/githubv4"
+)
+
+type LastRepo struct {
+	Name        string
+	Description string
+	CreatedAt   time.Time
+	UpdatedAt   time.Time
+}
+
+func (c LastRepo) String() string {
+	msg := fmt.Sprintf("\nThe last repo updated \n\n **%v** \n \t %v \n Created At: %s \n Updated At: %s", c.Name, c.Description, c.CreatedAt.Format("2006-01-02"), c.UpdatedAt.Format("2006-01-02"))
+	if len(c.Name) == 0 {
+		msg = fmt.Sprintf("Could not find a repository")
+	}
+
+	return msg
+}
+
+func (g *GithubService) GetLastUpdatedRepoByUsername(ctx context.Context, username string) (*LastRepo, error) {
+
+	options := GetContributionsByUsernameOptions{
+		Username: username,
+	}
+
+	lastRepo := &LastRepo{
+		Name:        "",
+		Description: "",
+		CreatedAt:   time.Now(),
+		UpdatedAt:   time.Now(),
+	}
+
+	var lastRepoQuery struct {
+		User struct {
+			Repositories struct {
+				Nodes []struct {
+					Name        string
+					Description string
+					CreatedAt   time.Time
+					UpdatedAt   time.Time
+				}
```

Comment:
damn.. wish i remembered to comment this in the first round review.. can you capture the repo url? would be sick to  make the repo name linkable in the outputted text.
`url` is a supported field on the [Repository](https://docs.github.com/en/graphql/reference/objects#repository)

Reply:
note in go, the suggested variable name would be `URL` ( I believe VSCode IDE will notify of this )

Reply:
I added url under the description! :)